src/traits.rs

```diff
@@ -871,6 +939,22 @@ macro_rules! prim_int_impl {
                 <$T>::rotate_right(self, n)
             }
 
+            fn arithmetic_shl(self, n: u32) -> Self {
+                ((self as $S) << n) as $T
```

Comment:
Ok, so for signed it is defined as arithmetic shift, this should also be stated IMHO in documentation of `num`. Also it works incorrectly for unsigned integers bigger than `isize::max_integer()` as `0b1000_0000u8.arithmetic_shr(1)` would return `0b1100_0000u8` and I would rather expect true arithmetic shift right so it should be `0b0100_0000u8`.